roadmap/implementers-guide/src/node/utility/pvf.md

```diff
@@ -0,0 +1,114 @@
+# PVF
+
+The `pvf` module is responsible for handling preparation and execution subtasks
+for PVF code blobs.
+
+## Entrypoint
+
+This crate provides a simple API. You first `start` the validation host, which
+gives you the [handle][ValidationHost] and the future you need to poll.
+
+Then using the handle the client can send three types of requests:
+
+(a) PVF pre-checking. This takes the PVF [code][Pvf] and tries to prepare it
+(verify and compile) in order to pre-check its validity.
+
+(b) PVF execution. This accepts the PVF [`params`][ValidationParams] and the PVF
+[code][Pvf], prepares (verifies and compiles) the code, and then executes PVF
+with the `params`.
+
+(c) Heads up. This request allows to signal that the given PVF may be needed
+soon and that it should be prepared for execution.
+
+The preparation results are cached for some time after they either used or was
+signaled in heads up. All requests that depends on preparation of the same PVF
+are bundled together and will be executed as soon as the artifact is prepared.
+
+## Priority
+
+PVF execution requests can specify the [priority][Priority] with which the
+given request should be handled. Different priority levels have different
+effects. This is discussed below.
+
+Preparation started by a heads up signal always starts with the background
+priority. If there is already a request for that PVF preparation under way the
+priority is inherited. If after heads up, a new PVF execution request comes in
+with a higher priority, then the original task's priority will be adjusted to
+match the new one if it's larger.
+
+Priority can never go down, only up.
+
+## Mitigating disputes
+
+### Retrying execution requests
+
+If the execution request fails during **preparation**, we will retry if it is
+possible that the preparation error was transient (i.e. it was of type
+`PrepareError::Panic`, `PrepareError::TimedOut`, or
+`PrepareError::DidNotMakeIt`). We will only retry preparation if another
+requests comes in after 15 minutes, to ensure any potential transient conditions
+had time to be resolved. We will retry up to 5 times. See
+`can_retry_prepare_after_failure`.
+
+If the actual **execution** of the artifact fails, we will retry once if it was
+an `InvalidCandidate::AmbiguousWorkerDeath` error, after a 1 second delay to
```

Comment:
Did we really go with 1 second? Andronik suggested 3 seconds back then.

Reply:
Oh oops, I saw that [this](https://github.com/paritytech/polkadot/pull/6235#discussion_r1016553054) was resolved so I didn't make the change. I should have read more closely, because it sounds like we do want to go with 3 seconds. I can raise a separate PR, LMK.